Add new display modes for accuracy counter

cdwcgt opened the pull request:
closes https://github.com/ppy/osu/issues/9336

Increase display mode will show the accuracy that will grow from 0% (ACHIEVEMENT(+) in maimai)

Decrease display mode will show the accuracy that will decrement from 100% (ACHIEVEMENT(-) in maimai)

https://user-images.githubusercontent.com/34775378/210070059-4f11c9f9-13f4-41a8-8f04-ae08558bbcd6.mp4

cdwcgt committed:
add new accuracy counter display

bdach commented on osu.Game/Screens/Play/HUD/GameplayAccuracyCounter.cs:
```diff
@@ -4,17 +4,47 @@
 {
     public abstract partial class GameplayAccuracyCounter : PercentageCounter
     {
+        [SettingSource("Accuracy Display Mode")]
+        public Bindable<AccuracyType> AccType { get; } = new Bindable<AccuracyType>();
```
please don't use abbreviations like "acc" in identifiers

bdach commented on osu.Game/Screens/Play/HUD/GameplayAccuracyCounter.cs:
```diff
@@ -4,17 +4,47 @@
+            }, true);
+        }
+
+        public enum AccuracyType
+        {
+            Current,
+            Increase,
+            Decrease
```
needs some sort of description rather than whatever this is. something like "rolling", "worst achievable", "best achievable" maybe

cdwcgt committed:
rename `AccType` to `AccuracyDisplay`

cdwcgt committed:
remove nullable disabled

cdwcgt committed:
add description

mk56-spn commented on osu.Game/Screens/Play/HUD/GameplayAccuracyCounter.cs:
```diff
@@ -1,20 +1,54 @@
+
+        public enum AccuracyType
+        {
+            [Description("Rolling")]
```
dk how i feel about this, id have no idea what rolling does without seeing the code, " classic" or "base on previous hits" something of that sort would feel better IMO but YMMV

frenzibyte replied:
I would say "Accumulative" as a good alternative, meaning the accumulation of judged hits to decide the accuracy.

bdach replied:
don't see how "accumulative" is better than "rolling". if both of those are too difficult to understand then i'd advocate for "classic" or "standard"

bdach committed:
Naming pass

bdach committed:
xmldoc pass

bdach committed:
Adjust minimum values

bdach committed:
Add test coverage

bdach committed:
Rearrange formula for maximum accuracy

Feels like it's easier to understand this way. The difference of the
maximum scoring values for the entire beatmap and the max values for the
part of the beatmap that has already been played represents the act of
filling the rest of the unjudged objects with maximum results.

bdach committed:
Move binding to `LoadComplete()`

bdach committed:
Fix counter initially rolling down from 100% to 0% in minimum achievable mode

bdach commented:
In the hopes that I can move this along I've ~~essentially rewritten this~~ applied a round of cleanups. @cdwcgt in the interest of making the review process of any future pulls you may want to submit for us both I would like you to pay special attention to: naming changes applied, changes in xmldoc, as well as the test coverage added. Hopefully this should give an inkling of the standards we generally expect.

I've also fixed a stray issue that caused the accuracy meter to initially roll down from 100% to 0% in minimum accuracy mode (https://github.com/ppy/osu/pull/21945/commits/6509d3538c203f41049218899b1dd0471dcfa385).

Going to need a second approve for this one. Not super sure about putting the new accuracies in `ScoreProcessor` as it's pretty central, but there is probably no better place for it.

peppy commented on osu.Game/Screens/Play/HUD/GameplayAccuracyCounter.cs:
```diff
@@ -1,20 +1,64 @@
+            [Description("Standard")]
+            Standard,
+
+            [Description("Maximum achievable")]
+            MaximumAchievable,
+
+            [Description("Minimum achievable")]
+            MinimumAchievable
```
@bdach thoughts on making this something like "Count up" and "Count down"? It might be marginally more understandable to the end user.

bdach replied:
I dunno, that seems worse to me. "Count up" or "down" relative to what?

peppy replied:
To be honest, I'm not sure if anyone would actually want "minimum achievable" as a display mode. So maybe removing that then having "Current" of "Max achievable" as the only two options?

mk56-spn replied:
> To be honest, I'm not sure if anyone would actually want "minimum achievable" 

Its the primary ( only? ) display mod in atleast some DDR games so theres def precedent for it no?  i can see some people appreciating it as an option idk

peppy replied:
Fair enough. I guess we'll go with the current terminology and see what the user reception is like.
